pkg/util/docker/containers.go

```diff
@@ -348,3 +373,24 @@ func correctMissingIPs(addrs []containers.NetworkAddress, hostIPs []string) []co
 	}
 	return correctedAddrs
 }
+
+// crossIPsWithPorts returns the product of a list of IP addresses and a list of ports
+func crossIPsWithPorts(addrs []containers.NetworkAddress, ports []nat.Port) []containers.NetworkAddress {
+	res := make([]containers.NetworkAddress, len(addrs)*len(ports))
+	c := 0
+
+	for _, addr := range addrs {
+		if len(ports) == 0 {
+			res = append(res, addr)
```

Comment:
do we want to have network tuples with IPs without ports ?

Reply:
This should not happen, as exposedPorts is based on ports used by the container. However if it does, I think it's better to have a NetworkAddress w/o a port than nothing. I could be wrong though, what do you say @leeavital ? (i'm going to merge anyway to move forward, but we can revisit if that's not the case)

Reply:
cc @smehmood